### PR TITLE
[18.09] Backport: tests need to count data sets and collections

### PR DESCRIPTION
**Pull request by bernt-matthias**
Would like to backport this since IUC uses 18.09 in their CI tests.

### CODE REVIEW

**Comment by mvdbeek** [comment]
https://jenkins.galaxyproject.org/job/docker-api/13595/testReport/junit/test.api.test_workflows/WorkflowsApiTestCase/test_optional_workflow_output/ looks like we need to update the state in https://github.com/galaxyproject/galaxy/blob/release_18.09/test/api/test_workflows.py#L2202. That might be handled automatically on dev now, which would explain why the tests passed there but not against the release branch.

**Comment by bernt-matthias** [comment]
My guess is that the new parameter of the tool needs to be specified. I tried this with my latest commit.

File: test/api/test_workflows.py [hunk 1]
```diff
@@ -2200,6 +2200,7 @@ def test_optional_workflow_output(self):
     state:
       produce_out_1: False
       filter_text_1: '1'
+      produce_collection: True
```
Comment (mvdbeek) [inline test/api/test_workflows.py]: I think you need to set this to `False` here. This is the default, but if `state` is provided in gxformat2 workflows missing parameters aren't updated on 18.09.

**Review by mvdbeek — APPROVED** [review]
Thanks @bernt-matthias !

**Pull request merged**